src/Action/SetObjectFieldValueAction.php

```diff
@@ -40,18 +42,35 @@ final class SetObjectFieldValueAction
      */
     private $validator;
 
-    public function __construct(Environment $twig, Pool $pool, $validator)
+    /**
+     * @var DataTransformerResolver
+     */
+    private $resolver;
+
+    /**
+     * @param ValidatorInterface      $validator
+     * @param DataTransformerResolver $resolver
+     */
+    public function __construct(Environment $twig, Pool $pool, $validator, $resolver = null)
     {
         // NEXT_MAJOR: Move ValidatorInterface check to method signature
         if (!($validator instanceof ValidatorInterface)) {
             throw new \InvalidArgumentException(
-                'Argument 3 is an instance of '.\get_class($validator).', expecting an instance of'
-                .' \Symfony\Component\Validator\Validator\ValidatorInterface'
+                'Argument 3 is an instance of '.\get_class($validator).', expecting an instance of '
+                .ValidatorInterface::class
             );
         }
+
+        // NEXT_MAJOR: Move DataTransformerResolver check to method signature
+        if (!($resolver instanceof DataTransformerResolver)) {
+            @trigger_error('Calling '.__METHOD__.' without a '.DataTransformerResolver::class.' as fourth argument is deprecated since 3.x and will be throw an error in 4.0.', E_USER_DEPRECATED);
```

Comment:
```suggestion
            @trigger_error(sprintf(
                'Passing other type than %s in argument 4 to %s() is deprecated since since sonta-project/admin-bundle 3.x and will throw %s exception in 4.0.',
                DataTransformerResolver::class,
                __METHOD__,
                \TypeError::class
            ), E_USER_DEPRECATED);
```

Reply:
I knew it ! I was sure you would have a better error message than mine 😁

Reply:
Deja vu. I will change the message, but explain to me about the coding style. As a contributor, i write code in accordance with the coding style adopted in your project ([1](https://github.com/sonata-project/SonataAdminBundle/blob/895fb26c3eac518757ef2151704d901d3b6d550d/src/Action/SetObjectFieldValueAction.php#L47-L50), [2](https://github.com/sonata-project/SonataAdminBundle/blob/51b8c2ed4eee84c5a2bcde76a150277b0d496783/src/Controller/HelperController.php#L73-L76), [3](https://github.com/sonata-project/SonataAdminBundle/blob/067c35119afafde37a44ca1ca99f52bf13f30b89/src/Admin/AbstractAdmin.php#L1427-L1430)). Why do you suggest using `sprintf()` function instead of concatenating strings?

Reply:
I know this: https://symfony.com/doc/current/contributing/code/standards.html
Exception and error message strings must be concatenated using sprintf;

Old code doesn't always respect the coding style.

Reply:
@VincentLanglet okay. I understood you.